drivers/counter/counter_mcux_pit.c

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) 2020 NXP Semiconductor INC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_kinetis_pit
+
+#include <drivers/counter.h>
+#include <fsl_pit.h>
+
+#define LOG_MODULE_NAME counter_pit
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_COUNTER_LOG_LEVEL);
+
+struct mcux_pit_config {
+	struct counter_config_info info;
+	PIT_Type *base;
+	bool enableRunInDebug;
+	pit_chnl_t pit_channel;
+	void (*irq_config_func)(struct device *dev);
+};
+
+struct mcux_pit_data {
+	counter_top_callback_t top_callback;
+	void *top_user_data;
+};
+
+static int mcux_pit_start(struct device *dev)
+{
+	const struct mcux_pit_config *config = dev->config_info;
+
+	PIT_EnableInterrupts(config->base, config->pit_channel,
+			     kPIT_TimerInterruptEnable);
+
+	PIT_StartTimer(config->base, config->pit_channel);
+
+	return 0;
+}
+
+static int mcux_pit_stop(struct device *dev)
```

Comment:
Hehe, I love this function name. :)
https://www.merriam-webster.com/dictionary/pit%20stop